<ui version="4.0" stdsetdef="1" >
  <author>Sebastian Trueg</author>
  <comment></comment>
  <exportmacro></exportmacro>
  <class>base_K3bSoxEncoderConfigWidget</class>
  <widget class="QWidget" name="Form1" >
    <property name="geometry" >
      <rect>
        <x>0</x>
        <y>0</y>
        <width>419</width>
        <height>201</height>
      </rect>
    </property>
    <layout class="QVBoxLayout" >
      <property name="margin" >
        <number>0</number>
      </property>
      <item>
        <widget class="QCheckBox" name="m_checkManual" >
          <property name="text" >
            <string>Manual settings (used for all file types)</string>
          </property>
        </widget>
      </item>
      <item>
        <layout class="QHBoxLayout" >
          <item>
            <spacer name="spacer2" >
              <property name="sizeHint" >
                <size>
                  <width>20</width>
                  <height>10</height>
                </size>
              </property>
              <property name="sizeType" >
                <enum>QSizePolicy::Fixed</enum>
              </property>
              <property name="orientation" >
                <enum>Qt::Horizontal</enum>
              </property>
            </spacer>
          </item>
          <item>
            <layout class="QGridLayout" >
              <item row="1" column="0" >
                <widget class="QLabel" name="textLabel2" >
                  <property name="text" >
                    <string>Sample rate:</string>
                  </property>
                  <property name="wordWrap" >
                    <bool>false</bool>
                  </property>
                </widget>
              </item>
              <item row="2" column="1" >
                <widget class="QComboBox" name="m_comboEncoding" >
                  <property name="enabled" >
                    <bool>false</bool>
                  </property>
                  <property name="whatsThis" stdset="0" >
                    <string>&lt;p>The sample data encoding is signed linear (2's complement), unsigned linear, u-law (logarithmic), A-law (logarithmic), ADPCM, IMA_ADPCM, GSM, or Floating-point.&lt;/p>
&lt;p>&lt;b>U-law&lt;/b> (actually shorthand for mu-law) and &lt;b>A-law&lt;/b> are the U.S. and international standards for logarithmic telephone sound compression. When uncompressed u-law has roughly the precision of 14-bit PCM audio and A-law has roughly the precision of 13-bit PCM audio. A-law and u-law data is sometimes encoded using a reversed bit-ordering (i.e. MSB becomes LSB).&lt;br> &lt;b>ADPCM &lt;/b> is a form of sound compression that has a good compromise between good sound quality and fast encoding/decoding time. It is used for telephone sound compression and places where full fidelity is not as important. When uncompressed it has roughly the precision of 16-bit PCM audio. Popular versions of ADPCM include G.726, MS ADPCM, and IMA ADPCM. It has different meanings in different file handlers. In .wav files it represents MS ADPCM files, in all others it means G.726 ADPCM. &lt;br> &lt;b>IMA ADPCM&lt;/b> is a specific form of ADPCM compression, slightly simpler and slightly lower fidelity than Microsoft's flavor of ADPCM. IMA ADPCM is also called DVI ADPCM.&lt;br> &lt;b>GSM&lt;/b> is a standard used for telephone sound compression in European countries and is gaining popularity because of its good quality. It is usually CPU intensive to work with GSM audio data.&lt;/p> &lt;p>&lt;em>Description based on the SoX manpage&lt;/em>&lt;/p></string>
                  </property>
                  <item>
                    <property name="text" >
                      <string>Signed Linear</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>Unsigned Linear</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>u-law (logarithmic)</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>A-law (logarithmic)</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>ADPCM</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>IMA_ADPCM</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>GSM</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>Floating-Point</string>
                    </property>
                  </item>
                </widget>
              </item>
              <item row="1" column="1" >
                <widget class="QLineEdit" name="m_editSamplerate" >
                  <property name="enabled" >
                    <bool>false</bool>
                  </property>
                  <property name="text" >
                    <string>14400</string>
                  </property>
                  <property name="whatsThis" stdset="0" >
                    <string/>
                  </property>
                </widget>
              </item>
              <item row="3" column="0" >
                <widget class="QLabel" name="textLabel4" >
                  <property name="text" >
                    <string>Data size:</string>
                  </property>
                  <property name="wordWrap" >
                    <bool>false</bool>
                  </property>
                </widget>
              </item>
              <item row="2" column="0" >
                <widget class="QLabel" name="textLabel3" >
                  <property name="text" >
                    <string>Data encoding:</string>
                  </property>
                  <property name="wordWrap" >
                    <bool>false</bool>
                  </property>
                </widget>
              </item>
              <item row="0" column="0" >
                <widget class="QLabel" name="textLabel1" >
                  <property name="text" >
                    <string>Channels:</string>
                  </property>
                  <property name="wordWrap" >
                    <bool>false</bool>
                  </property>
                </widget>
              </item>
              <item row="0" column="1" >
                <widget class="QComboBox" name="m_comboChannels" >
                  <property name="enabled" >
                    <bool>false</bool>
                  </property>
                  <item>
                    <property name="text" >
                      <string>1 (mono)</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>2 (stereo)</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>4 (quad sound)</string>
                    </property>
                  </item>
                </widget>
              </item>
              <item row="3" column="1" >
                <widget class="QComboBox" name="m_comboSize" >
                  <property name="enabled" >
                    <bool>false</bool>
                  </property>
                  <item>
                    <property name="text" >
                      <string>Bytes</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>16-bit Words</string>
                    </property>
                  </item>
                  <item>
                    <property name="text" >
                      <string>32-bit Words</string>
                    </property>
                  </item>
                </widget>
              </item>
            </layout>
          </item>
        </layout>
      </item>
      <item>
        <spacer name="spacer1" >
          <property name="sizeHint" >
            <size>
              <width>250</width>
              <height>16</height>
            </size>
          </property>
          <property name="sizeType" >
            <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="orientation" >
            <enum>Qt::Vertical</enum>
          </property>
        </spacer>
      </item>
    </layout>
  </widget>
  <layoutdefault spacing="6" margin="11" />
    <connections>
    <connection>
      <sender>m_checkManual</sender>
      <signal>toggled(bool)</signal>
      <receiver>m_comboChannels</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
    <connection>
      <sender>m_checkManual</sender>
      <signal>toggled(bool)</signal>
      <receiver>m_editSamplerate</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
    <connection>
      <sender>m_checkManual</sender>
      <signal>toggled(bool)</signal>
      <receiver>m_comboEncoding</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
    <connection>
      <sender>m_checkManual</sender>
      <signal>toggled(bool)</signal>
      <receiver>m_comboSize</receiver>
      <slot>setEnabled(bool)</slot>
    </connection>
  </connections>
</ui>
